src/core/Dropdown/Dropdown/Dropdown.tsx

```diff
@@ -664,34 +668,43 @@ class BaseDropdown extends Component<DropdownProps> {
 }
 
 const StyledDropdown = styled(
-  ({ theme, ...passProps }: DropdownProps & SuomifiThemeProp) => (
-    <BaseDropdown {...passProps} />
-  ),
+  <T extends string = string>({
+    theme,
+    ...passProps
+  }: DropdownProps<T> & SuomifiThemeProp) => <BaseDropdown {...passProps} />,
 )`
   ${({ theme }) => baseStyles(theme)}
 `;
 
-const Dropdown = forwardRef(
-  (props: DropdownProps, ref: React.RefObject<HTMLButtonElement>) => {
-    const { id: propId, ...passProps } = props;
-    return (
-      <SuomifiThemeConsumer>
-        {({ suomifiTheme }) => (
-          <AutoId id={propId}>
-            {(id) => (
-              <StyledDropdown
-                theme={suomifiTheme}
-                id={id}
-                forwardedRef={ref}
-                {...passProps}
-              />
-            )}
-          </AutoId>
-        )}
-      </SuomifiThemeConsumer>
-    );
-  },
-);
+const DropdownInner = <T extends string = string>(
+  props: DropdownProps<T>,
+  ref: React.RefObject<HTMLButtonElement>,
+) => {
+  const { id: propId, ...passProps } = props;
+  return (
+    <SuomifiThemeConsumer>
+      {({ suomifiTheme }) => (
+        <AutoId id={propId}>
+          {(id) => (
+            <StyledDropdown
+              theme={suomifiTheme}
+              id={id}
+              forwardedRef={ref}
+              {...passProps}
+            />
+          )}
+        </AutoId>
+      )}
+    </SuomifiThemeConsumer>
+  );
+};
+
+export const Dropdown = React.forwardRef(DropdownInner) as <
+  T extends string = string,
+>(
+  props: DropdownProps<T>,
+  ref: React.RefObject<HTMLButtonElement>,
+) => ReturnType<typeof DropdownInner>;
 
-Dropdown.displayName = 'Dropdown';
-export { Dropdown, DropdownProvider, DropdownConsumer };
+DropdownInner.displayName = 'Dropdown';
```

Comment:
Needed to do this to get it working also in the styleguidist; without this it could not show the props in page.

Reply:
Might be a good idea to add this as a comment in the code as well just to clarify. Would also come in handy when we encounter similar cases elsewhere.

Reply:
Comment is now added.